js/manager.js

```diff
@@ -1,149 +1,144 @@
-        );
-    }
-
-    display_view(msg, view, options) {
-        const el = options.el || this.el;
-        return Promise.resolve(view).then(view => {
-            pWidget.Widget.attach(view.pWidget, el);
-            view.on('remove', function() {
-                console.log('view removed', view);
-            });
-            return view;
-        });
-    }
-
-    loadClass(className, moduleName, moduleVersion) {
-        if (moduleName === '@jupyter-widgets/output') {
-            return Promise.resolve(outputWidgets).then(module => {
-                if (module[className]) {
-                    return module[className];
-                } else {
-                    return Promise.reject(
-                        `Class ${className} not found in module ${moduleName}`
-                    );
-                }
-            })
-        } else {
-            return super.loadClass(className, moduleName, moduleVersion)
-        }
-    }
-
-    callbacks(view) {
-        const baseCallbacks = super.callbacks(view)
-        return {
-            ...baseCallbacks,
-            iopub: { output: (msg) => this._onError.emit(msg) }
-        }
-    }
-
-    _create_comm(target_name, model_id, data, metadata) {
-        const comm = this.kernel.connectToComm(target_name, model_id)
-        if (data || metadata ) {
-            comm.open(data, metadata)
-        }
-        return Promise.resolve(new ShimmedComm(comm))
-    }
-
-    _get_comm_info() {
-        return this.kernel.requestCommInfo({ target: this.comm_target_name})
-            .then(reply => reply.content.comms)
-    }
-}
+/***************************************************************************
+* Copyright (c) 2018, Voila contributors                                   *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+import { RenderMimeRegistry, standardRendererFactories } from '@jupyterlab/rendermime';
+import { WidgetManager as JupyterLabManager } from '@jupyter-widgets/jupyterlab-manager/lib/manager';
+import { HTMLManager, requireLoader } from '@jupyter-widgets/html-manager';
+import { WidgetRenderer } from '@jupyter-widgets/jupyterlab-manager/lib/renderer';
+import * as output from '@jupyter-widgets/jupyterlab-manager/lib/output';
+import * as base from '@jupyter-widgets/base';
+import * as controls from '@jupyter-widgets/controls';
+
+if (typeof window !== "undefined" && typeof window.define !== "undefined") {
+    window.define("@jupyter-widgets/base", base);
+    window.define("@jupyter-widgets/controls", controls);
+    window.define("@jupyter-widgets/output", output);

```

Comment:
I would like to find a way to get rid of those lines if possible.
If I remove `window.define("@jupyter-widgets/base", base);` I cannot load the bqplot example which requires the resource from unpkg and needs an external `@jupyter-widgets/base`

Reply:
Yes, unfortunately this is still required for the bundle-based means of loading custom widgets.

Eventually, we will have voila templates that don't load AMD bundles and will not require this.

Maybe a good way to move towards a cleaner solution would be to move this out of this file and include it into a the nbconvert template.

Reply:
I think we can do that in a different PR ok?

Reply:
ok